elephant/statistics.py

```diff
@@ -1550,11 +1550,12 @@ def _epoch_no_spread(self):
             bin_shift = .5 / self.sampling_rate
             left_edges -= bin_shift
 
-            # Ensure that an epoch does not start before the minimum t_start.
-            # Note: all spike trains share the same t_start and t_stop.
-            if left_edges[0] < self.t_start:
-                left_edges[0] = self.t_start
-                durations[0] -= bin_shift
+        # Ensure that an epoch does not start before the minimum t_start.
+        # Note: all spike trains share the same t_start and t_stop.
+        if left_edges[0] < self.t_start:
+            left_edges[0] = self.t_start
+            durations[0] -= bin_shift
+
```

Comment:
Was this change of the indentation intentional?  Consider the first if-clause in connection with the else statement:
```python
if self.sampling_rate:
    ....
else:
            warnings.warn('No sampling rate specified. ' ...
```

Now we have 
```python
if left_edges[0] < self.t_start:
    ...
else:
            warnings.warn('No sampling rate specified. ' ...
```

Reply:
I agree with the assessment. The variable `bin_shift` is set in the first `if`-clause that checks for the sampling rate. Thus, if this clause  is not executed, `bin_shift` is not defined. However, the second `if`-clause that test for `left_edges` uses the variable `bin_shift`. This will through an exception if `self.sampling_rate` is `False`. Therefore, I also think that the if clause needs to be either shifted back to the original state -- or one must do something else if there are other issues beyond the syntactic analysis that initiated this change.

Reply:
My bad, the indentation was accidental